docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -699,5 +699,50 @@ class HttpServerExampleSpec extends WordSpec with Matchers
     //#discard-close-connections
   }
 
+  "dynamic routing example" in compileOnlySpec {
+    import akka.actor.ActorSystem
+    import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+    import akka.http.scaladsl.server.Directives._
+    import akka.http.scaladsl.server.Route
+    import akka.stream.ActorMaterializer
+    import spray.json.DefaultJsonProtocol._
+    import spray.json._
 
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+
+    //#dynamic-routing-example
+    case class MockDefinition(path: String, requests: Seq[JsValue], responses: Seq[JsValue])
+    implicit val format = jsonFormat3(MockDefinition)
+
+    var state = Map.empty[String, Map[JsValue, JsValue]]
+
+    // fixed route to update state
+    val fixedRoute: Route = post {
+      pathSingleSlash {
+        entity(as[MockDefinition]) { mock =>
+          val mapping = mock.requests.zip(mock.responses).toMap
+          state += mock.path -> mapping
```

Comment:
perhaps even `state = state + (mock.path -> mapping)`?

(your example is indeed equally safe, but that might not be obvious to people newer to Scala)